### PR TITLE
client: fix inode ll_ref reference count leak

### PR DESCRIPTION
**Pull request by sepia-liu**
Fixes: https://tracker.ceph.com/issues/47918

Signed-off-by: sepia-liu <liuwei_coder@163.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by batrick — APPROVED** [review]
Good catch!

**Comment by jtlayton** [comment]
I've looked over the code, but I don't see where the extra reference comes from. @sepia-liu , could you explain that in the patch description?

File: src/client/Client.cc [hunk 1]
```diff
@@ -6579,9 +6579,7 @@ int Client::_lookup(Inode *dir, const string& dname, int mask, InodeRef *target,
       int r = make_request(req, perms, &tmptarget, NULL, rand() % mdsmap->get_num_in_mds());
 
       if (r == 0) {
-	Inode *tempino = tmptarget.get();
-	_ll_get(tempino);
-	*target = tempino;
+	*target = tmptarget.get();;
```
Comment (batrick) [inline src/client/Client.cc:6582]:
```suggestion
	*target = std::move(tmptarget);
```

**Review by batrick — CHANGES_REQUESTED** [review]

**Comment by jtlayton** [comment]
> I've looked over the code, but I don't see where the extra reference comes from. @sepia-liu , could you explain that in the patch description?

NVM, I see now. The caller is not expected to _ll_put the resulting InodeRef, so we just don't need this period.

File: src/client/Client.cc (hunk 1 above)
Comment (batrick) [reply src/client/Client.cc:6582]: (This avoids an unnecessary atomic increment/decrement.)

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201024.032205

failures unrelated.

**Pull request merged**